### DISCUSSION TITLE
async callbacks

### DISCUSSION THREAD

**Discussion by kirkwaiblinger** [iter-tools/iter-tools#457] [Q&A] [answered]
It's not clear to me from the docs how/whether one might use an async callback for the methods in this library. Consider the following code from native ES that doesn't quite work:
`const badUrls = ['www.google.com', '<not a website>', ... ].filter(async (url) => { 
       const result = await fetch(url);
       return !result.ok
});`
The filter callback returns a (truthy) promise for all inputs, which is clearly not what is intended here. Would there be a paradigm using this library to achieve this?

**Comment by conartist6** [comment, marked as answer]
```js
const badUrls = await execPipe(
  ['www.google.com', '<not a website>', /* ... */ ],
  asyncFilter(async (url) => { /* ... */ })
  asyncToArray
)
```
